litestar/openapi/config.py

```diff
@@ -88,31 +106,128 @@ class OpenAPIConfig:
 
     :class:`Reference <litestar.openapi.spec.reference.Reference>` objects.
     """
-    root_schema_site: Literal["redoc", "swagger", "elements", "rapidoc"] = "redoc"
-    """The static schema generator to use for the "root" path of `/schema/`."""
-    enabled_endpoints: set[str] = field(
-        default_factory=lambda: {
-            "redoc",
-            "swagger",
-            "elements",
-            "rapidoc",
-            "openapi.json",
-            "openapi.yaml",
-            "openapi.yml",
-            "oauth2-redirect.html",
-        }
-    )
-    """A set of the enabled documentation sites and schema download endpoints."""
     operation_id_creator: OperationIDCreator = default_operation_id_creator
     """A callable that generates unique operation ids"""
     path: str | None = field(default=None)
-    """Base path for the OpenAPI documentation endpoints."""
+    """Base path for the OpenAPI documentation endpoints.
+
+    If no path is provided the default is ``/schema``.
+
+    Ignored if ``openapi_router`` is provided.
+    """
+    render_plugins: Sequence[OpenAPIRenderPlugin] = field(default=())
+    """Plugins for rendering OpenAPI documentation UIs."""
+    openapi_router: Router | None = None
+    """A router for serving OpenAPI documentation and schema files.
+
+    If provided, ``path`` is ignored.
+
+    Ignored if the deprecated ``openapi_controller`` is provided.
+
+    Not required, however may be passed to customize the configuration of the router used to serve
+    the documentation endpoints. For example, to add middleware or guards to the router.
```

Comment:
```suggestion
    """An optional router for serving OpenAPI documentation and schema files.

    If provided, ``path`` is ignored.

    This parameter is also ignored if the deprecated :class:`OpenAPIConfig <.openapi.OpenAPIConfig>` ``openapi_controller`` kwarg is provided.

    The ``openapi_router`` is not required, but it can be passed to customize the configuration of the router used to serve the documentation endpoints. For example, you can add middleware or guards to the router.
```
this was a little confusing to me, feel free to ignore because I am likely bad